docs/guides/jupyter.rst

```diff
@@ -0,0 +1,377 @@
+which can provide a great experience for tutorials, examples, and other types of technical content.
+There are a few extensions that allow integrating Jupyter and Sphinx,
+and this document will explain how to achieve some of the most commonly requested features.
+
+.. _Jupyter: https://jupyter.org/
+
+By default, `Sphinx only supports reStructuredText source files for
+documentation <https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-source_suffix>`_.
+Luckily, there are a few extensions to also allow Jupyter notebooks as source files.
+
+.. _nb-background:
+
+Background on the existing relevant extensions
+----------------------------------------------
+
+There are two main extensions that add support Jupyter notebooks as source files in Sphinx:
+nbsphinx_ and MyST-NB_. Although they have similar intent and basic functionality,
+there are some differences:
+
+- nsphinx uses `pandoc <https://pandoc.org/>`_ to convert the Markdown from Jupyter notebooks
+  to reStructuredText and then to docutils AST,
+  whereas MyST-NB uses `MyST-Parser`_ to directly convert the Markdown text to docutils AST.
+  Both Markdown flavors are mostly equal, but they have some differences.
+- Both nbsphinx and MyST-NB support canonical ``.ipynb`` notebooks.
+  nbsphinx can process any format supported by `jupytext`_,
+  whereas MyST-NB focuses on MyST Markdown Notebooks
+  (see :ref:`below <other-formats>` for more information about other notebook formats).
+- nbsphinx executes each notebook during the parsing phase,
+  whereas MyST-NB can execute all notebooks up front
+  and cache them with `jupyter-cache <https://jupyter-cache.readthedocs.io/>`_.
+  This can result in shorter build times when when notebooks are slightly modified
+  if using MyST-NB.
+- nbsphinx provides functionality to create thumbnail galleries,
+  whereas MyST-NB does not have such functionality at the moment
+  (see :ref:`below <notebook-galleries>` for more information about galleries).
+- The visual appearance of code cells and their outputs is slightly different:
+  nbsphinx renders the cell numbers by default,
+  whereas MyST-NB doesn't.
+
+.. _nbsphinx: https://nbsphinx.readthedocs.io/
+.. _MyST-NB: https://myst-nb.readthedocs.io/
+.. _MyST-Parser: https://myst-parser.readthedocs.io/
+.. _jupytext: https://jupytext.readthedocs.io/
+
+Using which one to use depends on your use case. As general recommendations:
+
+- If you want to use :ref:`other notebook formats <other-formats>`
+  or :ref:`generate a thumbnail gallery from your notebooks <notebook-galleries>`,
+  nbsphinx is the right choice.
+- If you will be focusing on MyST Markdown notebooks
+  or want to leverage a more optimized execution workflow
+  and a more streamlined parsing mechanism,
+  you should use MyST-NB.
+
+For the rest of this document we will focus on `nbsphinx`_
+because it is more flexible and offers some extra functionality,
```

Comment:
I would politely disagree 😜 again myst-nb supports all formats, and there is also things like glue: https://myst-nb.readthedocs.io/en/latest/use/glue.html.  Obviously I’m biased though lol

Reply:
I am obviously biased too, since I've been using nbsphinx for 4 years and I know it much better than MyST-NB :) I will keep the examples with nbsphinx but try to make it look more neutral.

Plus: I didn't know about glue!